drivers/led/led_gpio.c

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <drivers/gpio.h>
+#include <drivers/led.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(led_gpio, CONFIG_LED_GPIO_LOG_LEVEL);
+
+#define DT_DRV_COMPAT gpio_leds
+#define LED_DATA(node_id)                      \
+{                                              \
+	{                                      \
+		DT_GPIO_LABEL(node_id, gpios), \
+		DT_GPIO_PIN(node_id, gpios),   \
+		DT_GPIO_FLAGS(node_id, gpios)  \
```

Comment:
Tricky: not every GPIO specifier has a flags field in devicetree, although almost all of them do. It's still a portability bug for a generic driver like this one to assume there is a flags field.

For a way to handle this, see the FLAGS_OR_ZERO macro from samples/subsys/usb/hid-cdc/src/main.c. I will try to implement https://github.com/zephyrproject-rtos/zephyr/issues/25827 after reviewing this to pull out the generic code if that helps.

Reply:
> Tricky: not every GPIO specifier has a flags field in devicetree

https://github.com/zephyrproject-rtos/zephyr/pull/27076 resolves this comment; feel free to ignore this comment as I expect this will be merged